src/core/thread/mle.cpp

```diff
@@ -4418,6 +4429,53 @@ void Mle::DelayedResponseMetadata::RemoveFrom(Message &aMessage) const
     OT_UNUSED_VARIABLE(error);
 }
 
+#if OPENTHREAD_CONFIG_TMF_DUA_REGISTRATION_ENABLE
+void Mle::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefixState aState)
+{
+    const otIp6Prefix *prefix;
+
+    if (aState == BackboneRouter::Leader::kDomainPrefixRemoved ||
+        (aState == BackboneRouter::Leader::kDomainPrefixRefreshed))
+    {
+        Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+    }
+
+    prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
+
+    VerifyOrExit(prefix != NULL, OT_NOOP);
+
+    if (aState == BackboneRouter::Leader::kDomainPrefixRefreshed)
+    {
+        Ip6::Address address = mDomainUnicastAddress.GetAddress();
+
+        mDomainUnicastAddress.GetAddress().Clear();
+        mDomainUnicastAddress.GetAddress().SetPrefix(prefix->mPrefix.mFields.m8, prefix->mLength);
+        mDomainUnicastAddress.GetAddress().SetIid(address.GetIid());
+        mDomainUnicastAddress.mPrefixLength = prefix->mLength;
+
+        Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
```

Comment:
Do we need to remove it before adding it here in "refreshed case"

Reply:
We also need to somehow track if the "add case" did add it before or not? e.g., can `clear()` the address to indicate this.